include/uv/unix.h

```diff
@@ -322,10 +322,19 @@ typedef struct {
   uv_idle_cb idle_cb;                                                         \
   void* queue[2];                                                             \
 
+#if defined(UV_PREFER_WAIT)
 #define UV_ASYNC_PRIVATE_FIELDS                                               \
   uv_async_cb async_cb;                                                       \
   void* queue[2];                                                             \
   int pending;                                                                \
+  uv_mutex_t mutex;                                                           \
+  uv_cond_t not_busy;
+#else
```

Comment:
Problem: this changes the ABI (grows `sizeof(uv_async_t)`), which means it's a backwards-incompatible change that can't land as-is in the v1.x branch. Is there a way to make this work without changing the struct size?

Small aside: ideally also without adding new defines / build-time options to the public headers. Adding defines to e.g. src/unix/internal.h is perfectly fine.

Reply:
🤔 Let us think a bit on if there is a way we can make it without the struct change. At first glance, think it's necessary. That would also mean we should move this to master instead of 1.x. 

Cleaned up the headers as requested.

Reply:
Moving this just to master should be fine too. We can potentially later backport an inefficient version of it that uses extra malloc calls to pack the extra data into a single word.